### DISCUSSION TITLE
两个过期值解决办法

### DISCUSSION THREAD

**Discussion by wxzvim** [nshen/learn-neovim-lua#76] [Ideas]
![image](https://user-images.githubusercontent.com/110797126/183543284-27186b7d-8314-42e4-a28e-ecae46cd0e7a.png)

`sed -i 's/client.resolved_capabilities/client.server_capabilities/g' $(grep -rl 'client.resolved_capabilities' *)`

![image](https://user-images.githubusercontent.com/110797126/183543401-b183c062-3059-4ace-a369-71ac093a368d.png)
`sed -i 's/vim.lsp.buf.formatting_sync/vim.lsp.buf.format/g' $(grep -rl 'vim.lsp.buf.formatting_sync' *)`

**Comment by nshen** [comment]
这个应该是 nvim 0.7 和 0.8 的区别，稳定版 0.7 应该没有提示，可以改成这样试试好不好用。

https://github.com/nshen/learn-neovim-lua/blob/f879437e92d14992937abacaeec14770e37ece28/lua/lsp/common-config.lua#L13-L19

**Reply by wxzvim** [reply]
厉害了，，这个肯定行，，有个疑问learn-neovim-lua/lua/lsp/common-config.lua这个文件教程没有说，直接建立这个文件就行了嘛？不需要加引用吗？

**Reply by nshen** [reply]
我就是觉得几乎所有lsp都要设置这个，就抽取出来了，肯定要引用的。你也可以复制过去，不用一模一样
https://github.com/nshen/learn-neovim-lua/blob/f879437e92d14992937abacaeec14770e37ece28/lua/lsp/config/lua.lua#L33-L36
